src/Microsoft.ML.Experimental/DataLoadSave/Database/DatabaseLoaderCursor.cs

```diff
@@ -441,9 +465,17 @@ private ValueGetter<VBuffer<double>> CreateVBufferDoubleGetterDelegate(ColInfo c
 
                     foreach (var seg in segs)
                     {
-                        for (int columnIndex = seg.Min; columnIndex < seg.Lim; columnIndex++, i++)
+                        if (seg.Name is null)
+                        {
+                            for (int columnIndex = seg.Min; columnIndex < seg.Lim; columnIndex++, i++)
+                            {
+                                editor.Values[i] = DataReader.IsDBNull(columnIndex) ? default : DataReader.GetDouble(columnIndex);
```

Comment:
Needs to stay `double.NaN`.

Same for the `else` code below.

Maybe we should write a test to ensure we don't break this and `float`.

Reply:
Fixed.

Adding tests will require a new database with null column types.